CLI_COMPREHENSIVE_DOC.md

```diff
@@ -0,0 +1,190 @@
+# Twilio CLI Comprehensive Document
```

Comment:
Let's rename the file as architecture.md since that might better indicate what the file is about and we don't need the word CLI in the name as it's anyway sitting in the CLI repo.

Reply:
Done